content/post/20200426-kubernetes-the-universal-control-plane/index.md

```diff
@@ -0,0 +1,113 @@
+
+## A Universal Control Plane
+
+<!-- Control Planes are foundational but underdiscussed - Kubernetes changing this by making control planes easier to build; Why control planes (AWS) - control theory, feedback loops, reactive systems, automation through services -->
+If the Kubernetes API design is *why* the project can be the ultimate infrastructure layer, control planes are *how* the vision is made into reality. Control planes are a foundational building block of designing modern cloud computing, underpinning infrastructure offerings spanning virtual machines, containers, cloud functions, databases, buckets, DNS, etc. For their importance to modern cloud infrastructure, I find that in practice they are not discussed often among cloud practitioners. I think it is because control planes are an implementation detail when building new kinds of infrastructure. Users are shielded from the complexities of managing control planes (thanks to declarative APIs) but they also lose out on exposure to the primitives cloud providers use to build their own services. This aspect is where Kubernetes is changing the state of play. Calling back to an earlier topic, Kubernetes provides the primitives for building control planes as **first-class abstractions**. Kubernetes democratizes infrastructure service management and creates a space where third-party cloud resources can be managed and used alongside community driven service implementations.
+
+### Defining Control Planes
+
+<!-- Control Plane - Came from networking, differs in cloud when data in much more complex; Why control planes (AWS) - control theory, feedback loops, reactive systems, automation through services  -->
+It is a good idea to take a step back and define what a control plane means here, especially in this newer context of software infrastructure. The original concept came out of [network routing](https://en.wikipedia.org/wiki/Control_plane) where the control plane tracks and manages the network topology and the packet routing rules while the data plane actively processes network requests. Marc Brooker [generalizes](https://brooker.co.za/blog/2019/03/17/control.html) the concept further:
```

Comment:
I would definitely define Control Planes before the Universal Control Plane section. I was so confused I had no idea what was being written about until I got here. I would clean it up and rearrange to define first and explain the universal control plane K8 connection after